lib/tracecontext/tracestate.js

```diff
@@ -0,0 +1,346 @@
+    const map = this.toMap()
+    const obj = {}
+    for (const key of map.keys()) {
+      obj[key] = map.get(key)
+    }
+    return obj
+  }
+
+  toMap () {
+    const map = new Map()
+
+    // first, serialize values from the internal representation. This means
+    // The W3C spec dictates that mutated values need to be on
+    // the left of the new trace string
+    if (Object.keys(this.values).length) {
+      map.set(this.listMemberNamespace, this._serializeValues(
+        this.values
+      ))
+    }
+    for (let i = 0; i < this.buffer.length; i++) {
+      const byte = this.buffer[i]
+      if (byte === 0) {
+        const indexOfKeyLength = i + 1
+        const indexOfKey = i + 2
+        const lengthKey = this.buffer[indexOfKeyLength]
+
+        const indexOfValueLength = indexOfKey + lengthKey
+        const indexOfValue = indexOfValueLength + 1
+        const lengthValue = this.buffer[indexOfValueLength]
+
+        const key = this.buffer.slice(indexOfKey, indexOfKey + lengthKey).toString()
+        const value = this.buffer.slice(indexOfValue, indexOfValue + lengthValue).toString()
+
+        map.set(key, value)
+
+        // skip ahead
+        i = indexOfValue + lengthValue - 1
+        continue
+      }
+    }
+
+    return map
+  }
+
+  toString () {
+    return this.toW3cString()
+  }
+
+  toW3cString () {
+    const json = this.toObject()
+    const chars = []
+    for (const [key, value] of Object.entries(json)) {
+      if (!value) { continue }
+      chars.push(key)
+      chars.push('=')
+      chars.push(value)
+      chars.push(',')
+    }
+    chars.pop() // remove final comma
+    return chars.join('')
+  }
+
+  _serializeValues (keyValues) {
+    const chars = []
+    for (const [key, value] of Object.entries(keyValues)) {
+      chars.push(`${key}:${value}`)
+      chars.push(';')
+    }
+    chars.pop() // last semi-colon
+    return chars.join('')
+  }
+
+  _validateVendorKey (key) {
+    if (key.length > 256 || key.length < 1) {
+      return false
+    }
+
+    const re = new RegExp(
+      '^[abcdefghijklmnopqrstuvwxyz0123456789_\\-\\*/]*$'
+    )
+    if (!key.match(re)) {
+      return false
+    }
+    return true
+  }
+
+  _validateElasicKeyAndValue (key, value) {
+    // 0x20` to `0x7E WITHOUT `,` or `=` or `;` or `;`
+    const re = /^[ \][!"#$%&'()*+\-./0123456789<>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ^_abcdefghijklmnopqrstuvwxyz{|}~]*$/
+
+    if (!key.match(re) || !value.match(re)) {
+      return false
+    }
+
+    if (key.length > 256 || value.length > 256) {
+      return false
+    }
+
+    return true
+  }
+
+  static fromBinaryFormatHexString (string, listMemberNamespace = 'es') {
+    return new TraceState(Buffer.from(string, 'hex'), listMemberNamespace)
+  }
+
+  static fromStringFormatString (string = '', listMemberNamespace = 'es') {
+    // converts string format to byte format
+    const bytes = []
+
+    const parts = string.split(',')
+    for (const [, part] of parts.entries()) {
```

Comment:
May be just use part.values?

Reply:
I presume that's a suggestion to use `parts.values()` in the `for` -- if I've misread that let me know :) 

Good callout -- that said the choice to use `entries` here (and elsewhere) was deliberate/explicit.  I've found it's simpler to 

1. _Always_ use either `entries()` or `Object.entries()`  
2. with the const `[...]` form
3. with a `for of` loop

The main benefit is with your loops more consistent/explicit you don't need to stop and think about what's happening in a particular loop. i.e. "What's being looped over" or "is a single variable the index or the value", etc.

Going to resolve this, but my preference is a minor one.  If there's a compelling reason to use `values()` (perf., etc)  I'm happy to switch this up.